tests/fsharpqa/Source/Conformance/SpecialAttributesAndTypes/Imported/CallerInfo/CallerMemberName.fs

```diff
@@ -45,6 +47,8 @@ type MyTy() =
             |> List.head
         MyTy.Check(result, Some("CheckMembers"), "lambda")
         MyTy.Check(functionVal (), Some("functionVal"), "functionVal")
+        MyTy.Check(anonymousLambda, Some("anonymousLambda"), "anonymousLambda")
+        MyTy.Check(asyncVal, Some("asyncVal"), "asyncVal")
```

Comment:
Certainly ``asyncVal`` result seems wrong. If the thing being defined by ``let x = ...`` is not a function then it doesn't seem sensible to be using the name ``x``.  

You should see similar dubious results for ``let x = if 1 > 0 then MyTy.GetCallerMemberName() else ""``

What do you get for 
```
let f () = 
    let x = MyTy.GetCallerMemberName() 
    x
```
or
```
let f () = 
    let x = 
        let g () = MyTy.GetCallerMemberName() 
        g()
    x
```
???

For ``anonymousLamba`` there is a quirk in the F# language spec where ``let x = fun args -> ...`` is considered identical to ``let x args = ...`` for module-defined function values.  So it may be you need to use an expression-bound definition.